app/api/user/allowance/route.ts

```diff
@@ -0,0 +1,45 @@
+import checkAddressBalances from "@/lib/hypersub/checkAddressBalances";
+import getUser from "@/lib/neynar/getNeynarUser";
+import getUserByUsername from "@/lib/neynar/getNeynarUserByUsername";
+import supabase from "@/lib/supabase/serverClient";
+import { NextRequest } from "next/server";
+
+export async function GET(req: NextRequest) {
+  const fid = req.nextUrl.searchParams.get('fid');
+  const username = req.nextUrl.searchParams.get('username');
+
+  try {
+    let user: any = null;
+    if (fid) user = await getUser(Number(fid));
+    if (username) user = await getUserByUsername(username);
+    if (!user) throw Error('username or fid required');
+
+    const allBalances = await checkAddressBalances(user.verifications);
+
+    let hyperSub = false;
+    for (const balance of allBalances) {
+      if (balance.error) throw balance.error;
+
+      hyperSub = !!balance.result;
+      if (hyperSub) break;
+    }
+
+    const { data } = await supabase.from('tips').select('*').eq('fid', user.fid);
```

Comment:
again, `optional` for now, but as we move forward
- avoid `supabase` queries at all cost
- use `stack` for more accurate query results regarding `tipping` and `points` on sonata